### ISSUE TITLE
Errors in during make on macOS Catalina 

### ISSUE THREAD

**Issue by dralism** [highperformancecoder/scidavis-old#160]
<! Please file bug reports at: https://sourceforge.net/p/scidavis/scidavis-bugs/> 

**Describe the bug**
I am following the instruction on install.md on macOS catalina using ports, all works fine until I reach step 7 (make) I got 5 errors and 45 warnings 

**Errors**
```
src/ApplicationWindow.cpp:5179:38: error: ordered comparison between pointer and zero
      ('const void *' and 'int')
         if (labels.contains(newName)>(0)){
             ~~~~~~~~~~~~~~~~~~~~~~~~^~~~
src/ApplicationWindow.cpp:7855:57: error: ordered comparison between pointer and zero
      ('const void *' and 'int')
                                if (onPlot.count() > 0 && plots.contains(w->name()) <= 0 )
                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~
src/ApplicationWindow.cpp:7861:96: error: ordered comparison between pointer and zero
      ('const void *' and 'int')
  ...((((Graph3D*)w)->formula()).contains(name,Qt::CaseSensitive) && plots.contains(w->name())<=0)
                                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~^ ~
src/ApplicationWindow.cpp:7880:30: error: ordered comparison between pointer and zero
      ('const void *' and 'int')
                        if (tables.contains(tl[0])<=0)
                            ~~~~~~~~~~~~~~~~~~~~~~^ ~
src/ApplicationWindow.cpp:10109:54: error: ordered comparison between pointer and zero
      ('const void *' and 'int')
        else if (fList[1].contains("(Z)",Qt::CaseSensitive) > 0)
```

**Expected behavior**
I am expecting to be able to complete the make process with no errors 

**Desktop (please complete the following information):**
 - OS: macOS Catalina
 - Browser Opera
 - Version 72

I would appreciate your thoughts and solution please

Reagrds

**Comment by dralism** [comment]
Hi again 
I tried repeating the steps on a fresh copy, now I got 2 errors and 127 warnings. No matter if I use qt4 or qt5 

```
In file included from /opt/local/include/qwt/qwt_plot.h:18:
/opt/local/include/qwt/qwt_scale_map.h:92:5: error: unknown type name
      'QT_STATIC_CONST'
    QT_STATIC_CONST double LogMin;
    ^
/opt/local/include/qwt/qwt_scale_map.h:93:5: error: unknown type name
      'QT_STATIC_CONST'
    QT_STATIC_CONST double LogMax;

```
Regards

**Comment by highperformancecoder** [comment]
Are you using qwt5? SciDaVis has not been ported to Qwt6.

On Thu, Dec 17, 2020 at 04:30:34PM -0800, dralism wrote:
> Hi again
> I tried repeating the steps on a fresh copy, now I got 2 errors and 127
> warnings. No matter if I use qt4 or qt5
> 
> In file included from /opt/local/include/qwt/qwt_plot.h:18: /opt/local/include/
> qwt/qwt_scale_map.h:92:5: error: unknown type name 'QT_STATIC_CONST'
> QT_STATIC_CONST double LogMin; ^ /opt/local/include/qwt/qwt_scale_map.h:93:5:
> error: unknown type name 'QT_STATIC_CONST' QT_STATIC_CONST double LogMax;
> 
> Regards
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.*
> 

-- 

----------------------------------------------------------------------------
Dr Russell Standish                    Phone 0425 253119 (mobile)
Principal, High Performance Coders     hpcoder@hpcoders.com.au
                      http://www.hpcoders.com.au
----------------------------------------------------------------------------

**Comment by dralism** [comment]
Thanks for your reply. 
Apologies maybe I am missing some details. I thought I am using qt5 as I did 
`port install qt5 `
Then I changed this `PATH=/opt/local/bin:/opt/local/libexec/qt4/bin:$PATH` to `PATH=/opt/local/bin:/opt/local/libexec/qt5/bin:$PATH`
Also it seems I have gwt ver 6.1.5 but I was not able to switch between qwt 5.2 and qwt 6.1.5 by using
` port select qwt qwt615` sure enough  this is invalid
I assumed I am using qt5 but it appears that I am wrong. 

I read through some other posts in issues section, apologies I posted this issue without prior reading. I learned that SciDavis 2.x is not ready yet to be compiled on mac, also given that there has not been huge changes as compared to ver 1.26, other than qt5 and python 3 support. 
I would stick to the pre-compiled ver 1.26 for the time being until new solutions come up. I do appreciate all the time and effort invested in this project. Thanks very much indeed. 

Regards
